Sources/NIO/CircularBuffer.swift

```diff
@@ -79,6 +85,80 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
         }
     }
 
+    /// Replaces the specified subrange of elements with the given collection.
+    ///
+    /// - Parameter subrange:
+    /// The subrange of the collection to replace. The bounds of the range must be valid indices of the collection.
+    ///
+    /// - Parameter newElements:
+    /// The new elements to add to the collection.
+    ///
+    /// *O(n)* where _n_ is the length of the new elements collection if the subrange equals to _n_
+    ///
+    /// *O(m)* where _m_ is the combined length of the collection and _newElements_
+    public mutating func replaceSubrange<C>(_ subrange: Range<Index>, with newElements: C) where C : Collection, E == C.Element {
+        precondition(subrange.lowerBound >= self.startIndex && subrange.upperBound <= self.endIndex, "Subrange out of bounds")
+
+        if subrange.count == newElements.count {
+            for (index, element) in zip(stride(from: subrange.lowerBound, to: subrange.upperBound, by: 1), newElements) {
```

Comment:
The `stride` here is not necessary, the `Range` is a sequence of the correct type already. This can just be `zip(subrange, newElements)`.

Reply:
Can't do it just like that because the compiler complains with:
```argument type 'Range<Int>' does not conform to expected type 'Sequence'```

I changed it to 
```swift
zip(subrange.lowerBound..<subrange.upperBound, newElements)
```

I think this reads better than using `stride`. Also added a comment pointing out why is not using just `zip(subrange, newElements)`